pkg/runtime/k0s/utils.go

```diff
@@ -99,3 +55,82 @@ func (k *Runtime) WaitK0sReady(ssh ssh.Interface, host net.IP) error {
 	}
 	return errors.New("failed to start k0s: failed to get k0s status after 10 seconds")
 }
+
+func GetClientFromConfig(adminConfPath string) (runtimeClient.Client, error) {
+	adminConfig, err := clientcmd.BuildConfigFromFlags("", adminConfPath)
+	if nil != err {
+		return nil, err
+	}
+
+	var ret runtimeClient.Client
+
+	timeout := time.Second * 30
+	err = wait.PollImmediate(time.Second*10, timeout, func() (done bool, err error) {
+		cli, err := runtimeClient.New(adminConfig, runtimeClient.Options{})
+		if nil != err {
+			return false, err
+		}
+
+		ns := corev1.Namespace{}
+		if err := cli.Get(context.Background(), runtimeClient.ObjectKey{Name: "default"}, &ns); nil != err {
+			return false, err
+		}
+
+		ret = cli
+
+		return true, nil
+	})
+
+	return ret, err
+}
+
+func (k *Runtime) WaitSSHReady(tryTimes int, hosts ...net.IP) error {
+	eg, _ := errgroup.WithContext(context.Background())
+	for _, h := range hosts {
+		host := h
+		eg.Go(func() error {
+			for i := 0; i < tryTimes; i++ {
```

Comment:
I think we can use time.Ticker here

Reply:
I also expect to try at the beginning, but this function is copied from k8s module, I am afraid of i will do something wrong when I use Ticker in a concurrence function.

Reply:
So let's put a TODO comment here, and then optimize it later